src/core/connection.c

```diff
@@ -1522,6 +1536,9 @@ QuicConnTryClose(
         if (ResultQuicStatus) {
             Connection->CloseStatus = (QUIC_STATUS)ErrorCode;
             Connection->CloseErrorCode = QUIC_ERROR_INTERNAL_ERROR;
+            if (ErrorCode != QUIC_STATUS_CONNECTION_IDLE) {
```

Comment:
I feel like we shouldn't include TIMEOUT in this case either. It's not a protocol error. It just means one side of the connection is no longer reachable.